src/camt/types.ts

```diff
@@ -114,6 +120,19 @@ export interface Transaction {
   returnAdditionalInformation?: string;
 }
 
+export interface BankTransactionCode {
+  /** Specifiesthe business area of the underlying transaction. */
+  domainCode?: string;
+  /** Specifies the family within the domain of the underlying transaction.  */
+  domainFamilyCode?: string;
+  /** Specifies the sub-product family within a specific family of the underlying transaction. */
+  domainSubFamilyCode?: string;
```

Comment:
Do you think we could make these enum/types? https://github.com/Svapnil/iso20022.js/commit/89412dabb2b8734064b40e2becc83bc59be2d2e6

Similar to this

Reply:
It is possible, but depending on how this should be implemented not within the `BankTransactionCode` interface. Here's the full details of the relations: https://www.iso20022.org/sites/default/files/2020-08/BTC_ExternalCodeListDescription_31Aug2020.doc 

Essentially, every domain code only allows certain FamilyCodes which in turn only allows some subFamilyCodes. If we want to be strict about it we could either create one type per Code X Family which has an enum for the sub family, or otherwise we can create one type per Code which has a number of families attached to them which have the enums. 

There are currently 11 domain codes and upward of 100 families, so probably the latter is better. Do you want to go for it?

Reply:
Yeah there's so many here, let alone the subfamilies..

Payments / CashManagement are probably the only domains that are relevant for this package so far. That said, it does include hundreds of subfamilies that we'd have to deal with. In the future I'm imagining something like:

```
export const DomainFamily = {
    Payments: {
        ...
    },
    CashManagement: {
        AccountBalancing: "ACCB",
        ....
    }
} as const;


export const SubdomainFamily = {
    Payments: {
        ...
    },
    CashManagement: {
        AccountBalancing: {
              CrossBorder: "XBRD",
              Sweeping: "SWEP"
       }
        ....
    }
} as const;
```

Basically, we'd codify https://www.iso20022.org/catalogue-messages/additional-content-messages/external-code-sets . I'll make this an Issue so it can be perhaps tackles in the future!